web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -515,51 +509,22 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         }
       }
     }
-    if (candidatePolygons.length === 1) {
-      this.zone.run(() => {
-        this.navigationService.selectLocationOfInterest(
-          candidatePolygons[0].get('id')
-        );
-      });
-      return;
-    }
-    this.openSelectLocationOfInterestDialog(candidatePolygons);
-  }
 
-  private openSelectLocationOfInterestDialog(
-    candidatePolygons: google.maps.Polygon[]
-  ) {
-    // TODO: Account for case where polygons in multipolygon overlap.
+    const candidatePolygonsByArea = candidatePolygons.sort(
```

Comment:
This function seems very business-logicy - can we move into the map service or a util? I think what we really want is a function which takes a collection of Ground Polygons and returns the smallest, e.gh. `getSmallestPolygon(polygons: Collection<Polygon>): Polygon?`